### PR TITLE
Guests with umbrellas no longer always avoid Maze when it rains

### PR DESCRIPTION
**Pull request by rik-smeets**
Since guests can use their umbrellas in the Maze, they should not always avoid going in the ride if they have one. They now have a 50% chance of entering the Maze - other guests will continue to look for a sheltered ride. 

Another minor change: guests on the Maze with an umbrella won't jump anymore, since they can't use their umbrella at that time and we all know guests really don't like to get wet from the rain. 

If there are any other ride types which allow guests to (continuously) use their umbrella on the ride, please let me know. Those rides can be given the new flag as well.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Is this actually something the community wants changed

**Comment by rik-smeets** [comment]
I don't know! Should I ask somewhere?

I simply noticed the default behaviour when playing the game and think this makes more sense. It isn't really a big change, but a nice touch in my opinion. The guests look good trying to pathfind their way through the maze with their umbrellas too.

**Comment by IntelOrca** [comment]
I think IRL, guests would still avoid a maze if it was raining. But you could just make it so that guests have a reduced chance of going on maze when it is raining but have umbrella rather than a zero chance.

**Comment by rik-smeets** [comment]
@IntelOrca I added a 50/50 chance, does that seem right?

I don't understand why some CI runs fail by the way, I can't relate it to my changes. Perhaps I'm missing something?

**Comment by IntelOrca** [comment]
@rik-smeets I am personally happy with that. Will have to see what others think.

The CI failures are test failures because you have altered the logic of the game, so some of the replays are now ending up with a different game state. This is to catch occasions when you unintentionally altered the game logic, but in this case it is intentional, so it should just be a case of updating the test expectation to the new checksum.

**Comment by rik-smeets** [comment]
@IntelOrca Ah, I see about the replay test expectations. I will update them whenever there's a green light on this pull request, interesting to hear others' thoughts.

File: src/openrct2/entity/Guest.cpp [hunk 1]
```diff
@@ -2591,6 +2592,16 @@ static void peep_update_ride_at_entrance_try_leave(Guest* peep)
     }
 }
 
+static bool peep_wants_to_use_ride_with_umbrella(Guest* guest, Ride* ride)
```
Comment (Gymnasiast) [inline src/openrct2/entity/Guest.cpp:2595]: We are moving away from functions and towards methods. Could you make this into a method of Guest?

Comment (IntelOrca) [reply src/openrct2/entity/Guest.cpp:2595]: I would change this to something like ShouldRideWhileRaining and make it do the shelter check too.

Comment (rik-smeets) [reply src/openrct2/entity/Guest.cpp]: Thanks for the suggestions! I made the requested changes :).

**Review by Broxzier — APPROVED** [review]

File: src/openrct2/entity/Guest.cpp [hunk 2]
```diff
@@ -2348,6 +2346,25 @@ int32_t Guest::GetParkEntryTime() const
     return ParkEntryTime;
 }
 
+bool Guest::ShouldRideWhileRaining(Ride* ride)
```
Comment (IntelOrca) [inline src/openrct2/entity/Guest.cpp:2349]:
```suggestion
bool Guest::ShouldRideWhileRaining(const Ride& ride)
```

File: src/openrct2/entity/Guest.cpp [hunk 3]
```diff
@@ -4807,7 +4824,7 @@ void Guest::UpdateRideMazePathfinding()
 
     if (IsActionInterruptable())
     {
-        if (Energy > 64 && (scenario_rand() & 0xFFFF) <= 2427)
+        if (Energy > 64 && (scenario_rand() & 0xFFFF) <= 2427 && !climate_is_raining())
```
Comment (duncanspumpkin) [inline src/openrct2/entity/Guest.cpp:4827]: Why this change? Is it just that jumping in the rain probably isn't realistic?

Comment (rik-smeets) [reply src/openrct2/entity/Guest.cpp:4827]: Yes, I noticed it looked a bit weird when the guests with umbrellas suddenly jumped (during which they don't hold them of course).

(This was already present before this change - they could have entered the maze before it started to rain)

**Review by duncanspumpkin — APPROVED** [review]

**Review by Gymnasiast — APPROVED** [review]

**Review by IntelOrca — APPROVED** [review]
